05_classify_mutations_multimodal/nbconverted/plot_multimodal_results.py

```diff
@@ -0,0 +1,304 @@
+print(compare_df.training_data.unique())
+compare_df[compare_df.gene == 'TP53'].head(10)
+
+
+# In[11]:
+
+
+sns.set({'figure.figsize': (10, 6)})
+
+plot_df = pd.DataFrame()
+
+# plot mean performance over all genes in pilot experiment
+for ix, gene in enumerate(results_df.identifier.unique()):
+    
+    plot_gene_df = compare_df[(compare_df.gene == gene)].reset_index(drop=True)
+    
+    # get the best-performing data types from the single-omics and multi-omics models
+    max_single_data_type = (
+        plot_gene_df[plot_gene_df.model_type == 'single-omics']
+          .groupby('training_data')
+          .agg('mean')
+          .delta_aupr.idxmax()
+    )
+    max_multi_data_type = (
+        plot_gene_df[plot_gene_df.model_type == 'multi-omics']
+          .groupby('training_data')
+          .agg('mean')
+          .delta_aupr.idxmax()
+    )
+    
+    # get samples with that data type
+    max_single_df = plot_gene_df[plot_gene_df.training_data == max_single_data_type]
+    max_multi_df = plot_gene_df[plot_gene_df.training_data == max_multi_data_type]
+    
+    # calculate difference between means and t-test p-val for that data type
+    mean_diff = max_single_df.delta_aupr.mean() - max_multi_df.delta_aupr.mean()
+    _, p_val = ttest_ind(max_single_df.delta_aupr.values,
+                         max_multi_df.delta_aupr.values)
+    print('{} diff: {:.4f} (pval: {:.4f})'.format(gene, mean_diff, p_val))
+    
+    plot_df = pd.concat((plot_df, max_single_df, max_multi_df))
+
+sns.boxplot(data=plot_df, x='gene', y='delta_aupr', hue='model_type')
+plt.title('Best performing single-omics vs. multi-omics data type, per gene')
+plt.xlabel('Target gene')
+plt.ylabel('AUPR(signal) - AUPR(shuffled)')
+
+
+# Summary and interpretation:
+# 
+# * In this last figure, we can see that the difference in distributions between the best-performing single-omics and multi-omics model is visually quite small for all of the target genes, and none of the differences are statistically significant using a t-test over the 8 total CV folds.
+# * Looking at the other two figures, when we consider all data types the difference between single-omics and multi-omics models is still pretty minimal in general. There are some genes where adding expression data helps a lot (EGFR, PIK3CA, KRAS) and others where it doesn't really (IDH1, SETD2), but in general there's not a lot of added predictive ability when expression is combined with other data types, suggesting that methylation may be most useful in that it correlates with expression patterns (although more follow-up would be needed to really confirm this).
+# * We had hypothesized that the reason for this might be that the multi-omics models are only selecting features from the best performing data type, but looking at the model coefficients this doesn't really seem to be the case; non-zero coefficients seem to be roughly evenly distributed between data types.
```

Comment:
More data != more better :(

Reply:
Yeah, I feel like there's a lot of room for finding more sophisticated methods that would do better here (like the sort of approaches you're looking at for combining datasets), but it was a bit surprising that this didn't work. What we're doing is super naive, but we still expected the added data to help somewhat.